ResearchKit/Common/ORKReviewStepViewController.m

```diff
@@ -0,0 +1,243 @@
+ 2.  Redistributions in binary form must reproduce the above copyright notice,
+ this list of conditions and the following disclaimer in the documentation and/or
+ other materials provided with the distribution.
+ 
+ 3.  Neither the name of the copyright holder(s) nor the names of any contributors
+ may be used to endorse or promote products derived from this software without
+ specific prior written permission. No license is granted to the trademarks of
+ the copyright holders even if such marks are included in this software.
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+ FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+#import "ORKReviewStepViewController.h"
+#import "ORKReviewStep.h"
+#import "ORKStep_Private.h"
+#import "ORKTaskViewController_Internal.h"
+#import "ORKStepViewController_Internal.h"
+#import "ORKSkin.h"
+#import "ORKTableContainerView.h"
+#import "ORKStepHeaderView_Internal.h"
+#import "ORKNavigationContainerView_Internal.h"
+#import "ORKTextChoiceCellGroup.h"
+
+
+typedef NS_ENUM(NSInteger, ORKReviewSection) {
+    ORKReviewSectionSpace1 = 0,
+    ORKReviewSectionAnswer = 1,
+    ORKReviewSectionSpace2 = 2,
+    ORKReviewSectionCount
+};
+
+@interface ORKReviewStepViewController () <UITableViewDataSource, UITableViewDelegate>
+
+@property (nonatomic, strong) ORKTableContainerView *tableContainer;
+@property (nonatomic, strong) UITableView *tableView;
+@property (nonatomic, strong) ORKStepHeaderView *headerView;
+
+@end
+
+@implementation ORKReviewStepViewController {
+    ORKNavigationContainerView *_continueSkipView;
+    ORKTextChoiceCellGroup *_choiceCellGroup;
+    ORKTextChoiceAnswerFormat *_answerFormat;
+}
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wobjc-designated-initializers"
+- (instancetype)initWithCoder:(NSCoder *)aDecoder {
+    self = [super initWithCoder:aDecoder];
+    return self;
+}
+#pragma clang diagnostic pop
+ 
+- (instancetype)initWithReviewStep:(ORKReviewStep *)reviewStep steps:(nullable NSArray *)steps resultSource:(nullable id<ORKTaskResultSource>)resultSource {
+    self = [self initWithStep:reviewStep];
+    if (self && [self reviewStep]) {
+        NSMutableArray *filteredSteps = [[NSMutableArray alloc] init];
+        [steps enumerateObjectsUsingBlock:^(id obj, NSUInteger idx, BOOL *stop) {
```

Comment:
So you mean: If someone passes in a standalone review step it should use the steps and resultSource that come enclosed, rather than the separately provided steps and resultSource that come with the init method?